acq4/util/imaging/contrast_ctrl.py

```diff
@@ -52,25 +51,33 @@ def zoomToImage(self):
         """
         self.imageItem.getViewBox().autoRange(items=[self.imageItem])
 
-    def levelsChanged(self):
-        if self.ui.btnAutoGain.isChecked() and not self.ignoreLevelChange:
-            if self.lastMinMax is None:
-                return
-            bl, wl = self.getLevels()
-            mn, mx = self.lastMinMax
-            rng = float(mx - mn)
-            if rng == 0:
-                return
-            newLevels = [(bl - mn) / rng, (wl - mn) / rng]
-            self.autoGainLevels = newLevels
+    def levelsOrLUTChanged(self):
+        if self.lastMinMax is None or not self.ui.btnAutoGain.isChecked() or self.ignoreLevelChange:
+            return
+        bl, wl = self.getLevels()
+        mn, mx = self.lastMinMax
+        rng = float(mx - mn)
+        if rng == 0:
+            return
+        newLevels = [(bl - mn) / rng, (wl - mn) / rng]
+        self.autoGainLevels = newLevels
+        self._cachedSaveName = None
 
     def alphaChanged(self, val):
-        self.alpha = val / self.ui.alphaSlider.maximum()  # slider only works in integers and we need a 0 to 1 value
+        self.alpha = val / self.ui.alphaSlider.maximum()  # slider only works in integers, and we need a 0 to 1 value
         self.imageItem.setOpacity(self.alpha)
 
     def getLevels(self):
         return self.ui.histogram.getLevels()
 
+    def save(self, img, dh: DirHandle) -> str:
+        if self._cachedSaveName is None:
+            lut = self.ui.histogram.getLookupTable(img)
```

Comment:
Instead just use `self.ui.histogram.saveState()` and this should be small enough to go in .index directly